File: challenge-016/veesh-goldman/perl5/ch-01.pl
#! /usr/bin/env perl

use v5.22;
#TASK: given a pie of pizza (computer people think of everything
#in terms of pizza) shared between 100 guests, where the first
#gets 1%, and the next gets 2% of what remains, etc: who gets the
#biggest piece?

my $guests = shift // 100;
my $biggest = {};

sub cut_a_slice_for {
  state $pie = 1;
  my $slice = (shift() / $guests ) * $pie;
  $pie -= $slice;
  return $slice
}

for my $guest ( 1 .. $guests ) {
  my $piece = cut_a_slice_for($guest); 
  $biggest = { name => $guest, size => $piece } if $piece > $biggest->{size}
}

say $biggest->{name}

File: challenge-230/steven-wilson/python/ch-01.py
#!/usr/bin/env python3


def separate_digits(*elements):
    return [int(char) for elem in elements for char in str(elem)]


if __name__ == "__main__":
    print(separate_digits(1, 34, 5, 6))
    print(separate_digits(1, 24, 51, 60))
